diff --color -Naru a/dist/adapters/drizzle-adapter/index.cjs b/dist/adapters/drizzle-adapter/index.cjs
--- a/dist/adapters/drizzle-adapter/index.cjs	2025-03-12 21:25:16.421608091 +1030
+++ b/dist/adapters/drizzle-adapter/index.cjs	2025-03-12 21:28:02.078023184 +1030
@@ -244,7 +244,7 @@
       const schemaModel = getSchema(model);
       const clause = where ? convertWhereClause(where, model) : [];
       const res = await db.select({ count: drizzleOrm.count() }).from(schemaModel).where(...clause);
-      return res.count;
+      return res[0].count;
     },
     async update(data) {
       const { model, where, update: values } = data;
diff --color -Naru a/dist/adapters/drizzle-adapter/index.mjs b/dist/adapters/drizzle-adapter/index.mjs
--- a/dist/adapters/drizzle-adapter/index.mjs	2025-03-12 21:25:16.421608091 +1030
+++ b/dist/adapters/drizzle-adapter/index.mjs	2025-03-12 21:28:10.672994032 +1030
@@ -242,7 +242,7 @@
       const schemaModel = getSchema(model);
       const clause = where ? convertWhereClause(where, model) : [];
       const res = await db.select({ count: count() }).from(schemaModel).where(...clause);
-      return res.count;
+      return res[0].count;
     },
     async update(data) {
       const { model, where, update: values } = data;
